### ISSUE TITLE
Prevent students from accidentally resubmitting assignments

### ISSUE THREAD

**Issue by ianalis** [jupyter/nbgrader#1171]
The Exchange currently does not stop accepting submissions when they are past the deadline and instead accepts multiple submissions of the same assignment even past due. Since the last submission is considered for grading, a student who accidentally submits past the deadline may be penalized even though their accidental submission is the same as the one they submitted before the deadline. Nbgrader should help students from accidentally resubmitting assignments and I have a few suggestions on how to do this:

- Add a prompt every time a student submits an assignment and proceed only if the student confirms it.
- Same as above but prompt only when they already have a submission for that assignment.
- Check whether the contents to be submitted are the same as those of the last submission and silently abort if they are.
- Same as above but with a warning message.
- When determining the submission that will be collected, pick the earliest submission with the same content as the last one.

This is an actual problem I have encountered and I'm happy to implement the preferred option based on this thread.

**Comment by Sefriol** [comment]
I think a prompt that just states "Submitting an assignment will replace any previous submission for this exercise. Do you want to proceed?" is enough. Anything more complicated than that would not be wise since there might be a real possibility that the exchange system is going to get replaced at some (SoonTM) point.

We had very similar discussion about this at our University and the conclusion we had was pretty much the statement above. Though, even this simple solution is quite low in our priority list.

**Comment by jhamrick** [comment]
Yes, I agree with @Sefriol 's assessment. I think a prompt is a good idea for now---potentially even with an additional warning to say if it is past the submission deadline (just so the student is aware).
